libraries/anvill_passes/src/InstructionFolderPass.cpp

```diff
@@ -523,11 +561,23 @@ bool InstructionFolderPass::FoldPHINodeWithCastInst(
     destination_type = instr->getDestTy();
   }
 
+  // NOTE(akshayk) : A phi node folding with cast instructions like IntToPtr
+  //                 PtrToInt, and BitCast may enter into an infinite loop
+  //                 if the folding instruction bounce between each other. Added
+  //                 checks to avoid such cases.
+  //
+  if ((cast_instr->getOpcode() == llvm::Instruction::IntToPtr) ||
```

Comment:
Lets remove this and just depend on not including them in the map / commenting them out in the map.